dynamiq/connections/connections.py

```diff
@@ -929,3 +930,30 @@ class DeepSeek(BaseApiKeyConnection):
 
     def connect(self):
         pass
+
+
+class Exa(Http):
+    """
+    Represents a connection to the Exa AI Search API.
+
+    Attributes:
+        type (Literal[ConnectionType.Exa]): The type of connection, which is always 'Exa'.
+        url (str): The URL of the Exa API.
+        api_key (str): The API key for authentication, fetched from the environment variable 'EXA_API_KEY'.
+        method (Literal[HTTPMethod.POST]): HTTP method used for the request, defaults to HTTPMethod.POST.
+    """
+
+    type: Literal[ConnectionType.Exa] = ConnectionType.Exa
+    url: str = Field(default="https://api.exa.ai/search")
+    api_key: str = Field(default_factory=partial(get_env_var, "EXA_API_KEY"))
+    method: Literal[HTTPMethod.POST] = HTTPMethod.POST
+
```

Comment:
```python3
class Exa(BaseApiKeyConnection):
    type: Literal[ConnectionType. Exa] = ConnectionType. Exa
    api_key: str = Field(default_factory=partial(get_env_var, "EXA_API_KEY"))
...
```

This is all we need.

Reply:
This would be inconsistent with the behavior of other tools. WDYT, @acoola?

Reply:
Other tools and their connections are poorly implemented. For example `TavilyTool` is [designed](https://github.com/dynamiq-ai/dynamiq/blob/b572490638fb200659a1dc7c1254c1fcaa8a9504/dynamiq/nodes/tools/tavily.py#L14) to work with an [exact endpoint](https://docs.tavily.com/docs/rest-api/api-reference#endpoint-post-search) that has an exact method + exact payload + exact response. Why would you want to overwrite the method or headers?

Reply:
The same is true for other tools. Overriding a `method` or an `URL` or `headers` only can break things.

Reply:
Here is the [endpoint](https://github.com/dynamiq-ai/dynamiq/blob/b572490638fb200659a1dc7c1254c1fcaa8a9504/dynamiq/nodes/tools/tavily.py#L126) we want to use. And this endpoint expects `POST`, not anything else.

Reply:
Regarding Tavily, if we want to add a new node for extraction or make changes to our existing node, we need to update the URL in the connection and use the same connection as we do with Cohere for LLMs and embedders.

https://docs.tavily.com/docs/rest-api/api-reference#endpoint-post-extract

Reply:
We will not have to update the URL (aka base URL), we will have to update the path only.

Reply:
And the path is defined on the node level, not in the connection.

Reply:
For example, the connection has two fields, `url` + `api_key` (this is more than enough), and even `api_key` should be enough.

`TavilySearchTool` will use `/search` path.
`TavilyExtractTool` will use `/extract` path.

Reply:
Our base HTTP API or HTTP connections depend on the `url` parameter.

Reply:
This is why we need to use `BaseApiKeyConnection` or `HttpApiKey`.

Reply:
We need to inherit from `Http` and define proper attributes that need for connection in connection